package.json

```diff
@@ -185,6 +185,8 @@
     "@kubernetes/client-node": "^0.15.1",
     "@sentry/electron": "^2.5.4",
     "@sentry/integrations": "^6.13.3",
+    "@ogre-tools/injectable": "^1.0.2",
```

Comment:
Did you look into alternatives for "injection library"? I haven't researched the options, but this dependency seems to have really low usage compared to e.g. https://www.npmjs.com/package/injection-js

Reply:
Also `injection-js` is written in typescript and has own types `*.d.ts`

Reply:
@ixrock Types for `injectable` are in this codebase for demonstrational purposes, mostly. The types are about to relocate to natural `injectable` soon enough.

Reply:
I am still not convinced that defaultProps (for class components) and object destructuring (for function components) is not enough for injection during testing.

Reply:
Why using default parameters/props is not good enough as pattern?

- Using default parameters means "bastard injection anti-pattern", as explained eg. here: https://joesusnick.medium.com/why-did-apple-advocate-for-bastard-injection-4b8268e91712
- Unit testability is not the only goal here. To achieve *loose coupling* and Inversion of Control would represent considerable value in codebase, right? :)

This is after using bastard injection *for years* before dropping it. I assure you we have the experience :) We are also putting our rep on the line to make all this worth it.

I feel there's a lot more to be said on this topic. We can take this to Slack if you want? :)

Reply:
> Be rational about where to apply it and what parts of your system you need to decouple. Don’t try to solve problems that you don’t have yet. Maybe you will never have them or the way that you solved them now will not fit when you really face them. At the end DI is just a means to an end, like any other pattern or technology that we use. It is not a goal itself.

_From here:_ https://ilya.puchka.me/dependency-injection-in-swift/

Yes, maybe we don't need yet DI everywhere for react component  `props` / `defaultProps`..?

Reply:
Absolutely! :)

Injectable should only be used when component needs dependencies, business logic or data etc.  

In most cases, where you think that you need to convert functional component to class component, you can greatly benefit from having business logic passed from outside of component and injectable is here to help with that. 

Class components are bit deeper discussion, since most times the requirement comes from the idea that you need business logic or lifecycle which should always be separated from stupid UI component. :)